docker/spack-stack/Dockerfile

```diff
@@ -0,0 +1,534 @@
+FROM ubuntu:20.04 as bootstrap
+
+ENV SPACK_ROOT=/opt/spack \
+    CURRENTLY_BUILDING_DOCKER_IMAGE=1 \
+    container=docker
+
+ENV DEBIAN_FRONTEND=noninteractive   \
+    LANGUAGE=en_US.UTF-8 \
+    LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8
+
+RUN apt-get -yqq update \
+ && apt-get -yqq upgrade \
+ && apt-get -yqq install --no-install-recommends \
+        build-essential \
+        ca-certificates \
+        curl \
+        file \
+        g++ \
+        gcc \
+        gfortran \
+        git \
+        gnupg2 \
+        iproute2 \
+        locales \
+        make \
+        python3 \
+        python3-pip \
+        python3-setuptools \
+        unzip \
+ && locale-gen en_US.UTF-8 \
+ && pip3 install boto3 \
+ && rm -rf /var/lib/apt/lists/*
+
+RUN mkdir $SPACK_ROOT && cd $SPACK_ROOT && \
+    git clone https://github.com/jcsda/spack . && git fetch origin spack-stack-1.5.1:container_branch && git checkout container_branch  && \
+    mkdir -p $SPACK_ROOT/opt/spack
+
+RUN ln -s $SPACK_ROOT/share/spack/docker/entrypoint.bash \
+          /usr/local/bin/docker-shell \
+ && ln -s $SPACK_ROOT/share/spack/docker/entrypoint.bash \
+          /usr/local/bin/interactive-shell \
+ && ln -s $SPACK_ROOT/share/spack/docker/entrypoint.bash \
+          /usr/local/bin/spack-env
+
+RUN mkdir -p /root/.spack \
+ && cp $SPACK_ROOT/share/spack/docker/modules.yaml \
+        /root/.spack/modules.yaml \
+ && rm -rf /root/*.* /run/nologin $SPACK_ROOT/.git
+
+# [WORKAROUND]
+# https://superuser.com/questions/1241548/
```

Comment:
Do we need to leave this comment in? Was this left in by mistake?

Reply:
This file is mostly generated by a spack-stack command, and so most of its contents and formatting is a result of the auto-generation process. However, I need to make a small manual change to it, so I've added the slightly modified version here. The only part that is not original is the specs: section. Given that this Dockerfile must be generated with spack-stack commands, it is not productive to waste time reformatting it. Every time spack-stack is updated and we have a new container, we'd have to make all those changes again. It might be better to completely automate the process entirely so that we create this Dockerfile on-the-fly with a script rather than saving a copy of it.